schema.sql

```diff
@@ -13,258 +18,301 @@ drop table if exists areas;
 drop table if exists attributes;
 drop table if exists data_types;
 drop table if exists time_series;
-drop table if exists piecewise_linear;
 drop table if exists transmission_interchange;
 drop table if exists entities;
-drop table if exists linkages;
-drop table if exists reserves;
+drop table if exists time_series_associations;
+drop table if exists time_series_metadata;
+drop table if exists single_time_series;
+drop table if exists deterministic_time_series;
+drop table if exists probabilistic_time_series;
 drop table if exists operational_data;
+drop table if exists supplemental_attributes;
+drop table if exists attributes_associations;
 
--- only generation units
-create table  generation_units (
-	unit_id integer primary key,
-	name text not null unique,
-	prime_mover text not null,
-	fuel_type text not null,
-	balancing_topology text not null references balancing_topologies(name),
-	start_year integer not null check (start_year >= 0),
-	rating float not null check (rating > 0 ),
-	base_power float not null check (base_power > 0),
-	check (base_power >= rating),
-	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
+-- Table of entities gets populated automatically once there is an insert on the core tables.
+-- NOTE: This table should not be interacted directly
+create table entities(
+        id integer primary key
+        ,entity_type text not null
+        ,entity_id integer not null
+        ,unique (id, entity_id, entity_type)
 );
 
--- only storge units
-create table  storage_units (
-	storage_unit_id integer primary key,
-	name text not null unique,
-	prime_mover text not null,
-	fuel_type text not null,
-	max_capacity float not null check (max_capacity > 0),
-	round_trip_efficiency float check (round_trip_efficiency >= 0),
-	balancing_topology text not null references balancing_topologies(name),
-	charging_efficiency float not null check (charging_efficiency > 0),
-	discharge_efficiency float not null check (discharge_efficiency > 0),
-	start_year integer not null check (start_year >= 0),
-	rating float not null default 1 check (rating > 0 ),
-	base_power float not null check (base_power > 0),
-	scenario text null,
-	check (base_power >= rating),
-	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
-);
+-- Table of generation units (aka, existing units).
+create table generation_units (
+   id integer primary key
+  ,name text not null
+  ,prime_mover text not null
+  ,fuel text null
+  ,balancing_topology text not null references balancing_topologies (name)
+  ,start_year integer not null check (start_year >= 0)
+  ,rating real check (rating > 0) default 1.0
```

Comment:
Is rating not in natural units?

Reply:
The original design had separated the rating and the base power. We could combine them.

Reply:
Rating and base power are different quantities. They shouldn't be mixed

Reply:
The difficulty here is that 1.0 implies that it is in per-unit. We should make this field equal to the base power if not defined under the assumption that all the fields in the db are in natural units

Reply:
I will add that change.